app/src/resource-planing/resource-planing.service.ts

```diff
@@ -200,4 +201,207 @@ export class ResourcePlaningService {
+                lastWeekNumber: lastWeekNumber,
+                totalWeeksQuantity: totalWeeksQuantity,
+                hoursPerWeek: hoursPerWeek * 3600000,
+                startDate: resource.start_date,
+                endDate: resource.end_date,
+                totalDuration: resource.total_duration,
+                projectName: resource.project_v2.name,
+            };
+        });
+
+        let weeks = this.splitPeriodToWeeks(startDate, endDate);
+        weeks = this.addWeekNumbers(weeks.weeksNormalized);
+
+        return this.distributeByWeekAndProject(resourceArr, weeks);
+    }
+
+    distributeByWeekAndProject(resourceArr: any, weeks: any) {
+        resourceArr.forEach(resource => {
+            weeks.forEach(week => {
+                if (week.weekNumber >= resource.firstWeekNumber && week.weekNumber <= resource.lastWeekNumber) {
+                    if (!week.planResources) {
+                        week.planResources = [];
+                    }
+                    week.planResources.push({
+                        timeForPlanResource: resource.hoursPerWeek,
+                        projectName: resource.projectName,
+                    });
+                }
+            });
+        });
+
+        return weeks;
+    }
+
+    distributeByWeek(resourceArr: any, weeks: any) {
+        resourceArr.forEach(resource => {
+            weeks.forEach(week => {
+                if (week.weekNumber >= resource.firstWeekNumber && week.weekNumber <= resource.lastWeekNumber) {
+                    if (!week.hoursPerWeek) {
+                        week.hoursPerWeek = 0;
+                    }
+                    week.hoursPerWeek = week.hoursPerWeek + resource.hoursPerWeek;
+                }
+            });
+        });
+
+        return weeks;
+    }
+
+    addWeekNumbers(weeks: any) {
+        weeks.forEach(week => (week.weekNumber = this.getWeekNumber(week.start)));
+        return weeks;
+    }
+
+    getWeekNumber(currentDate: string) {
+        let date = new Date(currentDate);
+        date.setHours(0, 0, 0, 0);
+        date.setDate(date.getDate() + 3 - ((date.getDay() + 6) % 7));
+        const week1 = new Date(date.getFullYear(), 0, 4);
+
+        return 1 + Math.round(((date.getTime() - week1.getTime()) / 86400000 - 3 + ((week1.getDay() + 6) % 7)) / 7);
+    }
+
+    getWeekPeriodByDate(date: any) {
+        // moment.locale('en', {
+        //     week: {
+        //         dow: 1, // Monday is the first day of the week.
+        //         doy: 7, // Sunday is the last day of the week.
+        //     },
+        // });
+        const startOfWeek = moment(date)
+            .startOf('week')
+            .startOf('day')
+            .format('YYYY-MM-DDTHH:mm:ss');
+        const endOfWeek = moment(date)
+            .endOf('week')
+            .endOf('day')
+            .format('YYYY-MM-DDTHH:mm:ss');
+
+        return {
+            start: startOfWeek,
+            end: endOfWeek,
+        };
+    }
```

Comment:
too
maybe it will be easier:
```
getWeekPeriodByDate(date: any) {
        const dateOfWeek = moment(date)
            .startOf('week')
            .startOf('day')
            .format('YYYY-MM-DDTHH:mm:ss');

        return {
            start: dateOfWeek,
            end: dateOfWeek,
        };
    }
```
?